### PR TITLE
Fix distributed join planning table swap

### PR DESCRIPTION
**Pull request by mxm**
The MergePhase creation code assumes that the right side always contains the smaller table which will be broadcasted. It swaps the relations/plans if that is not the case. We must not forget to swap back everything because otherwise we change the join sides. Also inverting the join type is not correct.

### CODE REVIEW

**Comment by matriv** [comment]
Please add a test and changelog entry.

**Comment by mxm** [comment]
@matriv done

File: CHANGES.txt [hunk 1]
```diff
@@ -24,6 +24,9 @@ Changes
 Fixes
 =====
 
+ - Fixed a bug which would incorrectly rewrite a join when table stats
```
Comment (matriv) [inline CHANGES.txt]: I guess it threw an error so I would rephrase it to: **Fixed a bug that caused an error to be thrown in case of a ``JOIN`` query when table stats indicated that the left table of a join is smaller than the right due to a wrong rewrite.**

File: sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java [hunk 2]
```diff
@@ -197,6 +197,18 @@ public void testLeftSideIsBroadcastIfLeftTableIsSmaller() throws Exception {
         assertThat(collect.collectPhase().distributionInfo().distributionType(), is(DistributionType.BROADCAST));
     }
 
+    @Test
+    public void testRightSideIsNotAlwaysNullIfLeftTableIsSmaller() {
```
Comment (matriv) [inline sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java]: How are we sure that **users** is smaller than **users_multi_pk**?

Comment (mxm) [reply sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java]: Have a look at the `prepare()` method. We also don't know in other tests. The goal is to test that we don't `null` a `MergePhase` when we have table stats enabled. That's a bit like testing a test but I can include an assertion for the table stats.

File: CHANGES.txt (hunk 1 above)
Comment (mxm) [reply CHANGES.txt]: No, it produced incorrect results.

Comment (matriv) [reply CHANGES.txt]: Then: **Fixed a bug that caused incorrect results to be returned in case of a ``JOIN`` query when table stats indicated that the left table of a join is smaller than the right due to a wrong plan rewrite.**

File: sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java (hunk 2 above)
Comment (matriv) [reply sql/src/test/java/io/crate/planner/consumer/NestedLoopConsumerTest.java]: Thx! didn't see that before, sorry.

File: CHANGES.txt (hunk 1 above)
Comment (mxm) [reply CHANGES.txt]: I think I'd skip the technical part of the join rewrite: **Fixed a bug that caused incorrect results to be returned for JOIN queries when the table stats indicated that the left table of a join is smaller than the right.**

**Comment by mxm** [comment]
@matriv Is this good to go now?

**Comment by mxm** [comment]
Ah, old tab. Thanks!
